build.sh

```diff
@@ -28,7 +28,14 @@ find_prunes() {
 }
 
 find_cmd() {
-  echo "find . -type f -and \( -perm +x -or -name '*.sh' \) $(find_prunes)"
+  # GNU `find` dropped compatibility support for the `+` syntax with 4.5.12 (release 2013), it
+  # instead uses `/` which has been supported for a long time. BSD `find` however still used `+`.
+  local perm_format_specifier="+"
+  if (( $(find --version /dev/null 2>&1 | grep GNU > /dev/null 2>&1; echo $?) == 0 )); then
```

Comment:
since `find --version` on BSD returns an error, couldn't we just write this as  ìf [ find --version ]; then GNU; else BSD; fi`?